core/src/main/kotlin/org/evomaster/core/languagemodel/LanguageModelConnector.kt

```diff
@@ -0,0 +1,241 @@
+        }
+
+        workerPool.submit(task)
+    }
+
+    /**
+     * @return answer for the prompt as [Prompt] if exists
+     * @return null if there is no answer for the prompt
+     */
+    fun getAnswerByPrompt(prompt: String): Prompt? {
+        return prompts.filter { it.value.prompt == prompt && it.value.hasAnswer() }.values.firstOrNull()
+    }
+
+    /**
+     * @return answer for the UUID of the prompt
+     */
+    fun getAnswerById(id: String): Prompt? {
+        return prompts[id]
+    }
+
+    /**
+     * To query the large language server with a simple prompt.
+     * @return answer string from the language model server
+     */
+    fun query(prompt: String): String? {
+        if (!config.languageModelConnector) {
+            return null
+        }
+
+        validatePrompt(prompt)
+
+        return makeQuery(prompt)
+    }
+
+    private fun makeQuery(prompt: String, id: String? = null): String? {
+        validatePrompt(prompt)
+
+        val languageModelServerURL = getLanguageModelServerURL()
+
+        val languageModelName = getLanguageModelName()
+
+        val requestBody = objectMapper.writeValueAsString(
+            OllamaRequest(
+                prompt = prompt,
+                stream = false,
+                model = languageModelName.toString()
+            )
+        )
+
+        val response = call(languageModelServerURL, requestBody)
+
+        if (!id.isNullOrBlank()) {
+            if (prompts.contains(id)) {
+                val existingPrompt = prompts[id]
+
+                if (existingPrompt != null) {
+                    existingPrompt?.answer = response
+                }
+            }
+        }
+
+        return response
+    }
+
+    /**
+     * Private method to make the call to the large language model server.
+     *
+     * Note: If you are using Ollama as a server, please make sure to set the
+     * CORS origin for Ollama on the host operating system.
+     *
+     * Reference:
+     * https://medium.com/dcoderai/how-to-handle-cors-settings-in-ollama-a-comprehensive-guide-ee2a5a1beef0
+     */
+    private fun call(languageModelServerURL: String, requestBody: String): String? {
+        try {
+            val connection = URL(languageModelServerURL).openConnection() as HttpURLConnection
+            connection.requestMethod = "POST"
+            // TODO: set to avoid long running calls
+            connection.connectTimeout = 4000
+            connection.setRequestProperty("Content-Type", "application/json")
+            connection.useCaches = false
+            connection.doInput = true
+            connection.doOutput = true
+
+            val writer = DataOutputStream(connection.outputStream)
+            writer.writeBytes(requestBody)
+            writer.flush()
+            writer.close()
+
+            if (connection.responseCode != HttpURLConnection.HTTP_OK) {
+                LoggingUtil.getInfoLogger().warn("Failed to connect to language model server.")
```

Comment:
in the log messages, should have the status code returned. 
also this should be the log of this class, and not the info-logger. 
also, would make sense to open a connection on service startup, and provide info-logger message if it fails (eg due to wrong provided URL)